wolfcrypt/src/sha.c

```diff
@@ -719,6 +719,14 @@ int wc_ShaFinal(wc_Sha* sha, byte* hash)
         return BAD_FUNC_ARG;
     }
 
+#ifndef WC_NO_HARDEN
+    /* We'll add a 0x80 byte at the end,
+    ** so make sure we have appropriate buffer length. */
+    if (sha->buffLen > WC_SHA_BLOCK_SIZE - 1) {
+        return BAD_FUNC_ARG;
```

Comment:
This is an internal error.
It shouldn't happen when the implementation is working appropriately.
The best error code I've found for this kind of issue is BAD_STATE_E.